packages/andromeda-modules/src/time_gate.rs

```diff
@@ -0,0 +1,101 @@
+use andromeda_std::{amp::AndrAddr, andr_exec, andr_instantiate, andr_query, error::ContractError};
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::{ensure, Addr};
+
+#[andr_instantiate]
+#[cw_serde]
+pub struct InstantiateMsg {
+    pub gate_addresses: Vec<AndrAddr>,
+    pub cycle_start_time: CycleStartTime,
+    pub time_interval: Option<u64>,
+}
+
+#[andr_exec]
+#[cw_serde]
+pub enum ExecuteMsg {
+    UpdateCycleStartTime { cycle_start_time: CycleStartTime },
+    UpdateGateAddresses { new_gate_addresses: Vec<AndrAddr> },
+    UpdateTimeInterval { time_interval: u64 },
+}
+
+#[andr_query]
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    #[returns(Addr)]
+    GetCurrentAdoPath {},
+    #[returns(CycleStartTime)]
+    GetCycleStartTime {},
+    #[returns(Vec<AndrAddr>)]
+    GetGateAddresses {},
+    #[returns(String)]
+    GetTimeInterval {},
+}
+
+#[cw_serde]
+pub struct CycleStartTime {
```

Comment:
What's the reasoning for not using `Timestamp` or `Expiry` here?

Reply:
Of course, we can use `Timestamp` or `Expiry`.
By the way, I thought that it is a good way to use more visual methods via year, month, day... than only milliseconds.
If you want `Timestamp` or `Expiry` for this, I will do so.

Reply:
For consistency let's use our `Expiry` type.

Reply:
resolved it.